### DISCUSSION TITLE
Unable to login to Solis cloud

### DISCUSSION THREAD

**Discussion by robconway** [hultenvp/solis-sensor#226] [General]
I am unable to login to Solis cloud despite following the info, and I have an inverter installed, so I am unclear at the error. Any ideas?

My log show:

Logger: custom_components.solis.soliscloud_api
Source: custom_components/solis/soliscloud_api.py:193 
Integration: Solis Inverter ([documentation](https://github.com/hultenvp/solis-sensor/), [issues](https://github.com/hultenvp/solis-sensor/issues)) 
First occurred: 07:42:06 (1 occurrences) 
Last logged: 07:42:06

No inverters found

**Comment by jaakleen** [comment]
Hello I got the same error and changed the following function in soliscloud_api.py

    async def _get_station_details(self, plant_id: str) -> dict[str, str] | None:
        """
        Fetch Station Details
        """

        params = {
            'id': plant_id
        }
        result = await self._post_data_json(PLANT_DETAIL, params)

        if result[SUCCESS] is True:
            jsondata : dict[str, str] = result[CONTENT]
            if jsondata['code'] == '0':
                return jsondata
            else:
                _LOGGER.info("%s responded with error: %s:%s",PLANT_DETAIL, \
                    jsondata['code'], jsondata['msg'])
                result = { SUCCESS: True, MESSAGE:'OK', STATUS_CODE:200, CONTENT:{'code': '0', 'msg': 'success', 'data': {'sno': plant_id, 'latitude':0.0, 'longitude':0.0, 'cityStr': ''}}}
                jsondata : dict[str, str] = result[CONTENT]
                return jsondata
        else:
            _LOGGER.info('Unable to fetch details for Station with ID: %s', plant_id)
        return None

**Reply by hultenvp** [reply]
If you do this, then you simply ignore the response from the "stationDetail" call by filling in a response yourself.

It means you miss some of the data returned by the server.

Please enable debug logging and report the output. Read the wiki if you don't know how to enable debug logging.

Cheers

**Comment by jaakleen** [comment]
I know! I logged a Call to Solis about this. The api was successfull but
return data has an issue so to not be Blocked I return the id as the
plant_name. They are looking for somebody to look at it for Two weeks Now.

Regards,

Jaak
Op vr 16 dec. 2022 om 19:06 schreef hultenvp ***@***.***>

**Comment by alienatedsec** [comment]
Please check #232
